Control protocol with session starting

RouNNdeL opened the pull request.

RouNNdeL committed:
Create ControlService, join control events with GUI

RouNNdeL committed:
Add owner information to LobbyUpdate packet

RouNNdeL committed:
Add missing isOwner lobby update argument to ScreenController

BWalusiak committed:
Add lobby update

BWalusiak reviewed (commented):
Broken connection on either side causes 100% CPU usage.

RouNNdeL commented on client/core/src/com/cyberbot/bomberman/screens/lobby/LobbyLayout.java:
```diff
@@ -117,35 +113,57 @@ public void clicked(InputEvent event, float x, float y) {
 
     public void updateLobby(Lobby lobby, boolean isOwner) {
-        // TODO: Update lobby
+
+        Table ui2 = new Table();
+        ui2.setDebug(false);
+
+        ui2.setPosition(1, 1);
+        ui2.setWidth(tableWidth);
+        ui2.setHeight(worldHeight);
+        addActor(ui2);
+
+        if (isOwner) {
+
+            TextButton startGameButton = new TextButton("Start Game", skin2);
+            setupButton(ui2, tableWidth, startGameButton);
+            startGameButton.addListener(new ClickListener() {
+                @Override
+                public void clicked(InputEvent event, float x, float y) {
+                    delegate.startGame();
+                }
+            });
+
+            ui2.add().height(10).row();
+        }
+        Label lobbyID = new Label(lobby.getId(), skin2);
+        lobbyID.setWidth(tableWidth);
+        lobbyID.setAlignment(1);
+        lobbyID.setFontScale(8);
+        lobbyID.setWrap(false);
+        ui2.add(lobbyID).width(tableWidth).height(playerLabelHeight).row();
+
+        ArrayList<Client> clients = lobby.getClients();
+        for (int i = 0; i < clients.size(); i++) {
+            addPlayer(clients.get(i).getNick(), i);
+        }
+        for (int i = 3; i > clients.size() - 1; i++) {
+            removePlayer(i);
+        }
```
The update should not create the whole layout from scratch

RouNNdeL commented on client/core/src/com/cyberbot/bomberman/screens/lobby/LobbyLayout.java:
```diff
@@ -117,35 +113,57 @@ public void clicked(InputEvent event, float x, float y) {
     public void removePlayer(int number) {
-        playerEmpty[number] = true;
         playerLabels[number].setText("Empty");
     }
```
Those methods seem useless, as we'll usually update from the full state, not calculate the delta.

BWalusiak committed:
Add login screen, optimize LobbyLayout

RouNNdeL committed:
Add login (no password yet)

RouNNdeL committed:
Fix ClientControlService high CPU

Caused by SocketException ignored due to google/gson#1717

RouNNdeL committed:
Improve inputs in LoginLayout

RouNNdeL committed:
Remove empty GameScreenManager

RouNNdeL committed:
Properly send username to the server, minor code improvements

RouNNdeL committed:
Rename server source folder to 'kotlin'

RouNNdeL committed:
Improve player login/registration

- When the player is not yet registered and will attempt to login, they will get registered automatically if their nick has not been used
- When the player is registered they will be logged in if the password matches

RouNNdeL committed:
Handle lobby leave and ownership transfer

RouNNdeL committed:
Remove LibGDX dependency from server

RouNNdeL committed:
Add server logging

RouNNdeL committed:
Improve TileMap loading and validation

RouNNdeL committed:
Basic only server simulated networked gameplay
